code/modules/mob/living/basic/space_fauna/space_dragon/space_dragon.dm

```diff
@@ -77,6 +77,11 @@
 	buffet = new(src)
 	buffet.Grant(src)
 
+/mob/living/basic/space_dragon/Destroy()
+	QDEL_NULL(fire_breath)
+	QDEL_NULL(buffet)
```

Comment:
You can just null these the actions will delete themselves if their mob is deleted